channel/app.go

```diff
@@ -90,18 +93,24 @@ type (
 		// parameters and version must be 0.
 		InitState(*Params, []Action) (Allocation, Data, error)
 
-		// DecodeAction decodes actions specific to this application. This has to be
-		// defined on an application-level because every app can have completely
-		// different actions; during decoding the application needs to be known to
-		// know how to decode an action.
-		DecodeAction(io.Reader) (Action, error)
+		// NewAction returns an instance of action specific to this
+		// application. This has to be defined on an application-level because
+		// every app can have completely different action. This instance should
+		// be used for unmarshalling the binary representation of the action.
+		//
+		// It is zero-value is safe to use.
```

Comment:
```suggestion
		// It is zero-value and safe to use.
```
Like that?

Reply:
I have rephrased it in 27f6382f . Does it look fine now ?